### PR TITLE
Add credit command

### PR DESCRIPTION
**Pull request by 733amir**
#91 : The credit command will print the past week and all-time contributors' usernames and number of commits. The lists are sorted by number of commits (descending).

### CODE REVIEW

File: internal/app/credit/contributors.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	commits, _, err := ghc.Repositories.ListCommits(
+		context.Background(),
+		"aidanaden",
+		"canvas-sync",
+		&github.CommitsListOptions{Since: since},
+	)
+
+	// If any error happened, print it and exit with code 1.
+	if err != nil {
+		fmt.Printf("err: %+v", err)
```
Comment (aidanaden) [inline internal/app/credit/contributors.go:27]: use `pterm.Error` to print errors instead, similar to rest of the app

File: internal/app/credit/contributors.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	contributors, _, err := ghc.Repositories.ListContributors(
+		context.Background(),
+		"aidanaden",
+		"canvas-sync",
+		nil,
+	)
+
+	// If any error happened, print it and exit with code 1.
+	if err != nil {
+		fmt.Printf("err: %+v", err)
```
Comment (aidanaden) [inline internal/app/credit/contributors.go:76]: use `pterm.Error` to print errors instead, similar to rest of the app

**Review by aidanaden — CHANGES_REQUESTED** [review]
logging nits: will update the command description after - thank u v much :)

File: cmd/credit.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        println()
+
+        // Top 500 contributors
+        pterm.FgCyan.Println("All contributors:")
+        allContributors := credit.Contributors()
+        for _, c := range allContributors {
+            pterm.FgCyan.Printfln("- %s", c)
+        }
+        println()
+	},
```
Comment (aidanaden) [inline cmd/credit.go:39]: is it possible to print the following instead?
- new contributors (first pr/commit within the last week)
- active contributors (top 10 contributors within the last month)

i think this'll incentivise both active development and new contributors to join

Comment (733amir) [reply cmd/credit.go:39]: I changed the code to calculate and print "active contributors". But the "new contributors" you said is not simple. Do you have any idea, how we should do it correctly and efficiently? There are no APIs that can do it. We can calculate it manually by receiving all the commits, but it is not efficient and can be impossible in the near future due to the increasing number of commits.

Comment (aidanaden) [reply cmd/credit.go:39]: i think using some of the the apis from https://docs.github.com/en/rest/metrics/statistics?apiVersion=2022-11-28 could be useful

**Comment by aidanaden** [comment]
closing due to stale

**Pull request closed without merging**